test/test_base_service.py

```diff
@@ -788,6 +788,166 @@ def test_retry_config_external():
     assert retry_err.value.reason == error
 
 
+@responses.activate
```

Comment:
Need to add some additional tests:
* redirect to same host (both inside cloud.ibm.com)
* redirect to same host (both outside cloud.ibm.com)

Also, because you are completely overriding the default "follow redirects" behavior, I think you'll need to add some tests for redirect scenarios involving POST and a representative sample of redirect status codes.   There are some subtle gotcha's when implementing redirects and we need to make sure that we are providing equivalent function as compared to the default "follow redirects" behavior.
I had to do this in the Java implementation as well.  For example, for 301/302 status codes, a POST should be re-written as a GET with no body (it may fail in the redirected server, but that's what the HTTP semantics should be).  On the other hand, 307/308 status codes should NOT cause a POST to be re-written as a GET.
You should study the code that you are overriding and try to match it as closely as possible other than adding support for the *.cloud.ibm.com stuff.

Reply:
Tests are already there those scenarios. Copied from the tests
```python
def test_redirect_ibm_to_ibm_success():
    url_from = 'http://region1.cloud.ibm.com/'
    url_to = 'http://region2.cloud.ibm.com/'
    ...
```
and
```python
def test_redirect_not_ibm_to_not_ibm_fail():
    url_from = 'http://region1.notcloud.ibm.com/'
    url_to = 'http://region2.notcloud.ibm.com/'
    ...
```
Are you talking about something else?

Reply:
> Also, because you are completely overriding the default "follow redirects" behavior...

That's not the case. Still the `requests` package handles the redirections, the `"allow_redirect": False"` option only prevents to fire the next request automatically, but it's still being processed and resolved appropriately. From the `session.py` in the `requests` package:
```python
# If redirects aren't being followed, store the response on the Request for Response.next().
        if not allow_redirects:
            try:
                r._next = next(
                    self.resolve_redirects(r, request, yield_requests=True, **kwargs)
                )
            except StopIteration:
                pass
```
Let's touch base later, to make sure we are on the same page! :)

Reply:
> Are you talking about something else?

1. redirect to same host (both inside cloud.ibm.com):
from: region1.cloud.ibm.com
to:   region1.cloud.ibm.com

2. redirect to same host (both outside cloud.ibm.com):
from: region1.notcloud.ibm.com
to: region1.notcloud.ibm.com

Redirecting to the same host SHOULD allow the safe headers to be included in the redirected request, regardless of whether the host is inside or outside the cloud.ibm.com domain.

Reply:
> Let's touch base later, to make sure we are on the same page!

Yes, let's discuss tomorrow.